languages/ruby/reference/exercise-concepts/nucleotide-count.md

```diff
@@ -0,0 +1,53 @@
+# Concepts of nucleotide-count
+
+## General
+
+- class level methods: used as the starting point of the exercise. They may be named with `self.[method_name]` notation or with a `self` block. Another way of creating class methods is extending a module.
+- module methods: depending on the approach, they may be used instead of the class methods.
+- factory pattern and object creation: the method `from_dna` may be used to create instances of the main class.
+- instance variables: store the common value/state for the instance.
+- accessors: `att_reader` may be used to access instance variables.
+- access modifiers: `private` may be used to restrict access to the readers.
+- method definition: the `histogram` method needs to be defined.
+- method arguments: the method `from_dna` needs an argument.
+- string: one string is passed as the input.
+- `chars`: may be used to convert the string into an array of characters.
+- strings mutation: `freeze` may be used to prevent string constants from mutating. The directive `frozen_string_literal: true` may also be used for the same purpose.
```

Comment:
Should mutability be mentioned, since the exercise is about doing a count, not a mutation that we are doing or not doing?

This almost doesn't seem like a list of concepts to me, but a list of methods that may be used, which is an implementation detail, more than a concept that is needed for solving this exercise.

The following seem more like a listing of concepts:

- module methods: depending on the approach, they may be used instead of the class methods.
- factory pattern and object creation: the method `from_dna` may be used to create instances of the main class.
- instance variables: store the common value/state for the instance.
- accessors: `att_reader` may be used to access instance variables.
- access modifiers: `private` may be used to restrict access to the readers.
- method definition: the `histogram` method needs to be defined.
- method arguments: the method `from_dna` needs an argument.

The listing of methods seem like indicating the potential built in methods that someone may use to solve the exercise.

Reply:
@kotp I do feel like "strings" and "chars" are definitely concepts this exercise depends on. Another way to look at it: which Concept Exercises should be created for this exercise to be solved by a student? I feel that the student should at least know about "strings" (I don't know if "chars" is a separate concept in Ruby), so to me that implies it must also be a concept.

The Concept that I feel should probably be rephrased is "strings mutation," as I feel that the Concept being referred to might actually be "Immutability" (freeze also works on other types, right?).

Reply:
Now I also think we should get rid of the mutation/mutability concept for this exercise because it's not necessary to solve the exercise. It's possible to solve it with Strings, freeze the constants is only a good practise here.

Reply:
"chars" is a programming name of something, "characters" being spelled out makes it better, inherently.  (Unless we are talking about fish, then I think that "char" is the plural and singular of that word.)